### ISSUE TITLE
Support ES6 (non-transpiled) stage classes

### ISSUE THREAD

**Issue by ViktorMikhalkin** [jsGiven/jsGiven#202]
Looks like I'm doing something wrong. Could you provide a simple example to run the tests against Protractor with Typescript. I've could not run the example from guide.

The code
```
import { scenario, scenarios, setupForRspec, Stage } from 'js-given';

function sum(a: number, b: number): number {
    return a + b;
}

setupForRspec(describe, it);

class SumStage extends Stage {
    number1: number;
    number2: number;
    result: number;

    a_number(value) {
        this.number1 = value;
        return this;
    }

    another_number(value) {
        this.number2 = value;
        return this;
    }

    they_are_summed() {
        this.result = this.number1 + this.number2;
        return this;
    }

    the_result_is(expectedResult) {
        expect(this.result).toEqual(expectedResult);
        return this;
    }
}

scenarios('sum', SumStage, ({ given, when, then }) => {
    return {
        two_numbers_can_be_added: scenario({}, () => {
            given()
                .a_number(1)
                .and()
                .another_number(2);

            when().they_are_summed();

            then().the_result_is(3);
        }),
    };
});

```
produces the error below
```
Failed: Class constructor SumStage cannot be invoked without 'new'
TypeError: Class constructor SumStage cannot be invoked without 'new'
    at new extendedClass (D:\Dev\Projects\csos-ui-test-automation\node_modules\js-given\index.js:1278:126)
    at ScenarioRunner.buildStage (D:\Dev\Projects\csos-ui-test-automation\node_modules\js-given\index.js:1288:28)
    at stageBuilder (D:\Dev\Projects\csos-ui-test-automation\node_modules\js-given\index.js:921:44)
    at UserContext.<anonymous> (D:\Dev\Projects\csos-ui-test-automation\node_modules\js-given\index.js:1072:45)
    at D:\Dev\Projects\csos-ui-test-automation\node_modules\jasminewd2\index.js:112:25
    at new ManagedPromise (D:\Dev\Projects\csos-ui-test-automation\node_modules\selenium-webdriver\lib\promise.js:1077:7)
    at ControlFlow.promise (D:\Dev\Projects\csos-ui-test-automation\node_modules\selenium-webdriver\lib\promise.js:2505:12)
    at schedulerExecute (D:\Dev\Projects\csos-ui-test-automation\node_modules\jasminewd2\index.js:95:18)
    at TaskQueue.execute_ (D:\Dev\Projects\csos-ui-test-automation\node_modules\selenium-webdriver\lib\promise.js:3084:14)
    at TaskQueue.executeNext_ (D:\Dev\Projects\csos-ui-test-automation\node_modules\selenium-webdriver\lib\promise.js:3067:27)
From: Task: Run it("Two numbers can be added") in control flow
    at UserContext.<anonymous> (D:\Dev\Projects\csos-ui-test-automation\node_modules\jasminewd2\index.js:94:19)
    at attempt (D:\Dev\Projects\csos-ui-test-automation\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:4297:26)
    at QueueRunner.run (D:\Dev\Projects\csos-ui-test-automation\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:4217:20)
    at QueueRunner.execute (D:\Dev\Projects\csos-ui-test-automation\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:4199:10)
    at Spec.queueRunnerFactory (D:\Dev\Projects\csos-ui-test-automation\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:909:35)
    at Spec.execute (D:\Dev\Projects\csos-ui-test-automation\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:526:10)
    at UserContext.fn (D:\Dev\Projects\csos-ui-test-automation\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:5340:37)
    at attempt (D:\Dev\Projects\csos-ui-test-automation\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:4297:26)
    at QueueRunner.run (D:\Dev\Projects\csos-ui-test-automation\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:4217:20)
    at QueueRunner.execute (D:\Dev\Projects\csos-ui-test-automation\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:4199:10)
From asynchronous test: 
Error
    at ScenarioRunner.testFunc (D:\Dev\Projects\csos-ui-test-automation\node_modules\js-given\index.js:1426:9)
    at D:\Dev\Projects\csos-ui-test-automation\node_modules\js-given\index.js:969:31
    at Array.forEach (<anonymous>)
    at D:\Dev\Projects\csos-ui-test-automation\node_modules\js-given\index.js:964:27
    at Array.forEach (<anonymous>)
    at Suite.<anonymous> (D:\Dev\Projects\csos-ui-test-automation\node_modules\js-given\index.js:955:41)
    at addSpecsToSuite (D:\Dev\Projects\csos-ui-test-automation\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:1107:25)
    at Env.describe (D:\Dev\Projects\csos-ui-test-automation\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:1074:7)
    at describe (D:\Dev\Projects\csos-ui-test-automation\node_modules\jasmine-core\lib\jasmine-core\jasmine.js:4399:18)

```

**Comment by flegall** [comment]
Thanks for reporting this !

This sounds more like a bug on jsGiven's side.
I'll try to reproduce it soon and provide a working protractor +TS example.

Which node.js version are you using ? Can you post the tsconfig.json ?

Regards,

Florent

**Comment by flegall** [comment]
I've reproduced the same error with the jest+typescript example by setting the "target" to "es6" in the "compilerOptions" of the tsconfig.json.

I think it's a bug on js-given side when it subclasses an ES6 stage class.

Are you using the "target": "es6" in the "compilerOptions" section of your tsconfig.json ?
If yes, as a workaround could you try using a configuration file similar to https://github.com/jsGiven/jsGiven/blob/master/examples/jest-typescript/tsconfig.json ?
Especially, not specifying a "target", which means compiling for es5.

Regards,

Florent

**Comment by ViktorMikhalkin** [comment]
Hi Florent!
The node version is 8.8.1 and this is my tsconfig.json:
{
  "compilerOptions": {
    "target": "ES2017",
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "module": "commonjs",
    "moduleResolution": "node",
    "inlineSourceMap": true,
    "declaration": false,
    "noImplicitAny": false,
    "outDir": "tmp"
  },
  "exclude": ["node_modules"]
}

**Comment by flegall** [comment]
The bug is this issue, the example is #204

**Comment by flegall** [comment]
Clearly it a jsGiven bug.

As a workaround I think that changing the target to "ES5" will fix the issue.

**Comment by flegall** [comment]
It's related to this babel issue : https://github.com/babel/babel/issues/6299

Currently jsGiven is using transpiled stage classes, moving to non-transpiled classes is a breaking change.

I think I'll provide two versions of the library:
- the main one will be the one with es6 native classes, (that will still be imported as 'js-given')
- the other will be with transpiled classes (that has to be imported as 'js-given/es5-classes')

**Comment by flegall** [comment]
This is fixed in JsGiven 0.1.7 !
